homeassistant/components/home_connect/select.py

```diff
@@ -28,7 +33,30 @@
     HomeConnectCoordinator,
 )
 from .entity import HomeConnectEntity
-from .utils import get_dict_from_home_connect_error
+from .utils import bsh_key_to_translation_key, get_dict_from_home_connect_error
+
+FUNCTIONAL_LIGHT_COLOR_TEMPERATURE_ENUM = {
+    bsh_key_to_translation_key(option): option
+    for option in (
+        "Cooking.Hood.EnumType.ColorTemperature.custom",
+        "Cooking.Hood.EnumType.ColorTemperature.warm",
+        "Cooking.Hood.EnumType.ColorTemperature.warmToNeutral",
+        "Cooking.Hood.EnumType.ColorTemperature.neutral",
+        "Cooking.Hood.EnumType.ColorTemperature.neutralToCold",
+        "Cooking.Hood.EnumType.ColorTemperature.cold",
+    )
+}
+
+AMBIENT_LIGHT_COLOR_TEMPERATURE_ENUM = {
+    **{
+        bsh_key_to_translation_key(option): option
+        for option in ("BSH.Common.EnumType.AmbientLightColor.CustomColor",)
+    },
+    **{
+        str(option): f"BSH.Common.EnumType.AmbientLightColor.Color{option}"
+        for option in range(1, 100)
+    },
+}
```

Comment:
I believe the UI the user will see will be "Custom1", "Custom2", etc? If that's correct should we make is "Custom 1", "Custom 2", etc.?

Reply:
Why do you believe that?
It should be "1", "2", "3", etc.

Reply:
I asked it as a question.  Since it's a numeric with a range of 1-100 then we should be model it as a Number rather than as a Select.

Reply:
Well, it looked more like an affirmation but with a question mark at the end... 😅
Sorry about the misunderstanding.
Anyway, I tested it and it is displayed as expected:
![ambient_light_color_options](https://github.com/user-attachments/assets/3edaad21-daff-4ed8-bcc2-0630eca1d54a)

Reply:
> Since it's a numeric with a range of 1-100 then we should be model it as a Number rather than as a Select.

This a debate @MartinHjelmare and me had on discord the other day before creating the PR
He argued the following:
> If it's a range where the number shows the relationship, like a brightness, it would be ok as a number. If it's just different colors it should be a select.

And because it seems to be a selector for different colors, I decided to go with the select entity

Reply:
Sounds good.  We don't need to rehash that decision. I'm fine with it.

Why does custom appear in the dropdown in the screenshot? Is that a selectable value? In which case it isn't a number as is can be 1-100 or custom.

Reply:
Yes, that is a selectable value. I don't understand your last question.
More info for this entity from https://api-docs.home-connect.com/settings/#ambient-light-color:
> Possible values
> 
> - Custom Color
>    Key: BSH.Common.EnumType.AmbientLightColor.CustomColor
> - Color 1 .. 99
>   Key: BSH.Common.EnumType.AmbientLightColor.Color[1 .. 99]

Reply:
Got it. Thanks for the explanation.

Reply:
No problem!